Core/Object Arts/Dolphin/IDE/Base/SmalltalkWorkspace.cls

```diff
@@ -902,46 +909,20 @@ messagesForToken: anAssociation startingWith: aString maxItems: anInteger
 	"Private - Answer the set of selectors that could potentially be sent as messages following
 	the specified token."
 
-	| ignoreCase icon |
+	| selectors class ignoreCase icon |
 	ignoreCase := self isAutoCompletionCaseInsensitive.
 	icon := Message icon.
-	^((self classForToken: anAssociation)
-		ifNil: 
-			[| allSelectors |
-			allSelectors := Smalltalk developmentSystem allSelectors.
-			aString isEmpty
-				ifTrue: [allSelectors size > anInteger ifTrue: [#()] ifFalse: [allSelectors]]
-				ifFalse: 
-					[| messages |
-					messages := Set new.
-					allSelectors do: 
-							[:eachName |
-							(eachName beginsWith: aString ignoreCase: ignoreCase)
-								ifTrue: 
-									[messages add: eachName.
-									messages size > anInteger ifTrue: [^#()]]].
-					messages]]
-		ifNotNil: 
-			[:class |
-			| messages |
-			messages := Set new.
-			class withAllSuperclassesDo: (aString isEmpty
-						ifTrue: 
-							[
-							[:eachClass |
-							messages addAll: eachClass selectors.
-							messages size > anInteger ifTrue: [^#()]]]
-						ifFalse: 
-							[
-							[:eachClass |
-							eachClass selectors do: 
-									[:eachSelector |
-									(eachSelector beginsWith: aString ignoreCase: ignoreCase)
-										ifTrue: 
-											[messages add: eachSelector.
-											messages size > anInteger ifTrue: [^#()]]]]]).
-			messages])
-			collect: [:each | ScintillaListItem text: each icon: icon]!
+	class := self classForToken: anAssociation.
+	class == UndefinedObject ifTrue: [class := nil].
```

Comment:
The motivation for this was the case where a _variable_ currently has the value `nil`, e.g. immediately after restarting a frame in the Debugger, or perhaps after setting a pool variable to nil in a workspace. In this case the nil-ness is likely temporary, and I've found it more useful to be offered everything rather than almost nothing. Good point about the literal case, though—I could make the special case narrower, so it only applies if the receiver is a variable, not the literal `nil`.